sky/cli.py

```diff
@@ -1801,7 +1844,45 @@ def status(all: bool, refresh: bool, ip: bool, show_spot_jobs: bool,
                 with ux_utils.print_exception_no_traceback():
                     raise ValueError('Querying IP address is not supported '
                                      'for local clusters.')
+
             head_ip = handle.external_ips()[0]
+            if show_endpoints:
+                cloud = handle.launched_resources.cloud
+                if not provision_lib.supports(repr(cloud), 'query_ports'):
+                    with ux_utils.print_exception_no_traceback():
+                        raise ValueError('Querying endpoints is not supported '
+                                         f'for {cloud}.')
+
+                config = common_utils.read_yaml(handle.cluster_yaml)
+                port_details = provision_lib.query_ports(
+                    repr(cloud), handle.cluster_name_on_cloud,
+                    handle.launched_resources.ports, config['provider'])
+
+                if endpoint is not None:
+                    if endpoint not in port_details:
+                        with ux_utils.print_exception_no_traceback():
+                            raise ValueError(
+                                f'Port {endpoint} not exposed yet.\n'
+                                'If the cluster was recently started, please retry after a while.\n'
+                                'If you believe this is an error, please file an issue.'
```

Comment:
We can remove this line since it generally applies everywhere :) Let's remove all instances of this line.
```suggestion
```

Reply:
Done.